### PR TITLE
Added components to initialize admin permissions for configured user

### PR DESCRIPTION
**Pull request by skabashnyuk**
Added components to initialize admin permissions for configured user
fixes https://github.com/eclipse/che/issues/6402

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

File: multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+  @Inject UserManager userManager;
+
+  @Inject PermissionsManager permissionsManager;
+
+  @Inject EventService eventService;
+
+  @Inject
+  @Named("che.admin.name")
+  String name;
```
Comment (garagatyi) [inline multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java]: it is not nullable. Does this PR work without any manual configuration?

File: multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import org.eclipse.che.api.core.notification.EventService;
+import org.eclipse.che.api.core.notification.EventSubscriber;
+import org.eclipse.che.api.user.server.UserManager;
+import org.eclipse.che.api.user.server.event.PostUserPersistedEvent;
+import org.eclipse.che.multiuser.api.permission.server.model.impl.AbstractPermissions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gran system permission for 'che.admin.name' user. If user already exist it'll happen on component
```
Comment (sleshchenko) [inline multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java]: Looks like `Grants` should be instead `Gran`.

File: multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java (hunk 1 above)
Comment (skabashnyuk) [reply multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java]: no. I'll uncomment default value

File: dockerfiles/init/manifests/che.env [hunk 3]
```diff
@@ -497,6 +497,9 @@ CHE_SINGLE_PORT=false
 #CHE_JDBC_MAX__IDLE=10
 #CHE_JDBC_MAX__WAIT__MILLIS=-1
 
+# User name of the admin for whom will be granted system permissions in che after first login.
```
Comment (sleshchenko) [inline dockerfiles/init/manifests/che.env]: Actually not only on the first login but on each tomcat start, no?

**Review by garagatyi — COMMENTED** [review]
Looks ok, please, consider adding tests

File: assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties [hunk 4]
```diff
@@ -8,72 +8,62 @@
...
 #     create a new workspace. The RAM slider is adjusted to this maximum value.
 che.limits.workspace.env.ram=16gb
-
 #     The length of time that a user is idle with their workspace when the system will
 #     suspend the workspace by snapshotting it and then stopping it. Idleness is the
 #     length of time that the user has not interacted with the workspace, meaning that
 #     one of our agents has not received interaction. Leaving a browser window open
 #     counts toward idleness.
 che.limits.workspace.idle.timeout=-1
-
```
Comment (sleshchenko) [inline assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties]: Why did you remove these empty lines? I like the previous kind of formatting.

File: dockerfiles/init/manifests/che.env [hunk 5]
```diff
@@ -497,6 +497,10 @@ CHE_SINGLE_PORT=false
 #CHE_JDBC_MAX__IDLE=10
 #CHE_JDBC_MAX__WAIT__MILLIS=-1
 
+# Grant system permission for 'che.admin.name' user. If the user already exists it'll happen on
+# component startup, if not - during the first login when user is persisted in the database.
+#CHE_ADMIN_USER_NAME=admin
```
Comment (garagatyi) [inline dockerfiles/init/manifests/che.env]: looks like `user` part is odd here

File: multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+ * Grant system permission for 'che.admin.name' user. If the user already exists it'll happen on
+ * component startup, if not - during the first login when user is persisted in the database.
+ *
+ * @author Sergii Kabashniuk
+ */
+@Singleton
+public class AdminPermissionInitializer implements EventSubscriber<PostUserPersistedEvent> {
+  private static final Logger LOG = LoggerFactory.getLogger(AdminPermissionInitializer.class);
+
+  @Inject UserManager userManager;
```
Comment (sleshchenko) [inline multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java]: Why all fields of these class have package private visibility? Wouldn't it better to make them private?

File: dockerfiles/init/manifests/che.env (hunk 5 above)
Comment (skabashnyuk) [reply dockerfiles/init/manifests/che.env]: yep. ```codenvy.admin.name``` we had in codenvy

File: multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+  public void grantSystemPermissions(String userId) {
+    // Add all possible system permissions
+    try {
+      AbstractPermissionsDomain<? extends AbstractPermissions> systemDomain =
+          permissionsManager.getDomain(SystemDomain.DOMAIN_ID);
+      permissionsManager.storePermission(
+          systemDomain.newInstance(userId, null, systemDomain.getAllowedActions()));
+    } catch (ServerException | NotFoundException | ConflictException e) {
+      LOG.warn(
```
Comment (sleshchenko) [inline multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializer.java]: I'd put exception as the second argument to `LOG.warn` method or use `{}` for formatting.

**Review by garagatyi — APPROVED** [review]
Looks ok, still would be better to have tests, but it is not critical.

File: assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties [hunk 8]
```diff
@@ -8,72 +8,62 @@
...
 che.limits.organization.workspaces.count=-1
 #     The maximum number of running workspaces that a single organization is allowed.
 #     If the organization has reached this threshold and they try to start an
 #     additional workspace, they will be prompted with an error message. The
 #     organization will need to stop a running workspace to activate another.
 che.limits.organization.workspaces.run.count=-1
+#####                             KEYCLOAK                                        #####
+#     Grant system permission for 'che.admin.name' user. If the user already exists it'll happen on
+#     component startup, if not - during the first login when user is persisted in the database.
+che.admin.name=admin
```
Comment (sleshchenko) [inline assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties]: I'm not sure that it should be in `KEYCLOAK` section. I'd put it below `CHE SYSTEM` section and rename this property to something like `che.system.admin_name`

File: assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties [hunk 9]
```diff
@@ -77,3 +77,7 @@ che.limits.organization.workspaces.count=-1
 #     additional workspace, they will be prompted with an error message. The
 #     organization will need to stop a running workspace to activate another.
 che.limits.organization.workspaces.run.count=-1
+#####                             CHE SYSTEM                                        #####
```
Comment (sleshchenko) [inline assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties]: we've already had `CHE SYSTEM` section above. Let's move it there

File: assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties [hunk 10]
```diff
@@ -77,3 +77,7 @@ che.limits.organization.workspaces.count=-1
...
+#     Grant system permission for 'che.admin.name' user. If the user already exists it'll happen on
+#     component startup, if not - during the first login when user is persisted in the database.
+che.admin.name=admin
```
Comment (sleshchenko) [inline assembly-multiuser/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties]: What about renaming property to `che.system.admin_name`?https://github.com/eclipse/che/pull/6414#discussion_r140727000

File: multiuser/api/che-multiuser-api-permission/src/test/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializerTest.java [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <R, T extends R> T cast(R qwe) {
+    return (T) qwe;
+  }
+
+  @Test
+  public void shouldAddSystemPermissionsOnPostUserPersistedEvent() throws Exception {
+    Injector injector = Guice.createInjector(new BaseModule());
```
Comment (sleshchenko) [inline multiuser/api/che-multiuser-api-permission/src/test/java/org/eclipse/che/multiuser/api/permission/server/AdminPermissionInitializerTest.java]: It would be much easier to use the injection into `AdminPermissionInitializer` with the constructor and avoid usage `Guice` here

**Review by garagatyi — APPROVED** [review]

File: dockerfiles/init/manifests/che.env [hunk 12]
```diff
@@ -497,6 +497,10 @@ CHE_SINGLE_PORT=false
 #CHE_JDBC_MAX__IDLE=10
 #CHE_JDBC_MAX__WAIT__MILLIS=-1
 
+# Grant system permission for 'che.system.admin_name' user. If the user already exists it'll happen on
+# component startup, if not - during the first login when user is persisted in the database.
+CHE_SYSTEM_ADMIN__NAME=admin
```
Comment (sleshchenko) [inline dockerfiles/init/manifests/che.env]: Shouldn't it be commented?

**Review by sleshchenko — APPROVED** [review]
LGTM
